File: toolkit/components/telemetry/TelemetryPing.js
/* -*- js-indent-level: 2; indent-tabs-mode: nil -*- */
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

Components.utils.import("resource://gre/modules/XPCOMUtils.jsm")
Components.utils.import("resource://gre/modules/Deprecated.jsm");

let JSM = {};
Components.utils.import("resource://gre/modules/TelemetryPing.jsm", JSM);

function TelemetryPing() {
  Deprecated.warning("nsITelemetryPing is deprecated. Please use TelemetryPing.jsm instead",
    "https://bugzilla.mozilla.org/show_bug.cgi?id=913070");
}

TelemetryPing.prototype = Object.create(JSM.TelemetryPing);
TelemetryPing.prototype.classID = Components.ID("{55d6a5fa-130e-4ee6-a158-0133af3b86ba}");
TelemetryPing.prototype.QueryInterface = XPCOMUtils.generateQI([Components.interfaces.nsITelemetryPing]);

this.NSGetFactory = XPCOMUtils.generateNSGetFactory([TelemetryPing]);

File: mobile/android/android-components/components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryPing.java
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.telemetry.ping;

import java.util.Map;

public class TelemetryPing {
    private final String type;
    private final String documentId;
    private final String uploadPath;
    private final Map<String, Object> measurementResults;

    /* package */ TelemetryPing(String type, String documentId, String uploadPath, Map<String, Object> measurementResults) {
        this.type = type;
        this.documentId = documentId;
        this.uploadPath = uploadPath;
        this.measurementResults = measurementResults;
    }

    public String getType() {
        return type;
    }

    public String getDocumentId() {
        return documentId;
    }

    public String getUploadPath() {
        return uploadPath;
    }

    public Map<String, Object> getMeasurementResults() {
        return measurementResults;
    }
}
